### PR TITLE
Adding Github Actions Workflow

### PR DESCRIPTION
**Pull request by vrr-21**
*Issue #, if available:*

*Description of changes:*
This change adds GitHub Actions Workflow to build using `rake` and to check code coverage using coveralls.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/ruby.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# This workflow uses actions that are not certified by GitHub.
```
Comment (mwjones-aws) [inline .github/workflows/ruby.yml:1]: Some newline inconsistencies in this file. Please be consistent about blank line use, and add a newline at the end of the file.

File: test/test_helper.rb [hunk 2]
```diff
@@ -1,3 +1,18 @@
+require 'simplecov'
```
Comment (mwjones-aws) [inline test/test_helper.rb:1]: These changes look identical to the changes in spec_helper.rb.

Comment (vrr-21) [reply test/test_helper.rb:1]: Yes, they are recommended by SimpleCov ([README](https://github.com/simplecov-ruby/simplecov#getting-started)). I can try using in only one if needed.

File: .github/workflows/ruby.yml (hunk 1 above)
Comment (mwjones-aws) [reply .github/workflows/ruby.yml:1]: Still some newline/indentation inconsistencies in this file.

**Comment by mwjones-aws** [comment]
I see this in the output of Run Tests on one of the checks. Let's turn off trying to submit the results to the coveralls API

```
 [Coveralls] Submitting to https://coveralls.io/api/v1
fatal: not a git repository (or any of the parent directories): .git
```

**Comment by vrr-21** [comment]
Turned off submitting results to coveralls API.

**Comment by vrr-21** [comment]
Continued in https://github.com/aws/aws-codedeploy-agent/pull/296

**Pull request closed without merging**